### ISSUE TITLE
segmentation fault

### ISSUE THREAD

**Issue by dennis605** [etexteditor/e#77]
I compiled both of the newest version under ubuntu 10.10 - debug and release.
Everything worked fine while building, but when i try to start "e" or "e.debug" i get a segmentation fault.
Strace throws this out:
<code>
open("/home/dennis/.icons/Macbuntu-Cursors/cursors/xterm", O_RDONLY) = 15
fstat(15, {st_mode=S_IFREG|0644, st_size=4160, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f64c5dea000
read(15, "Xcur\20\0\0\0\0\0\1\0\1\0\0\0\2\0\375\377 \0\0\0\34\0\0\0$\0\0\0"..., 4096) = 4096
lseek(15, 0, SEEK_SET)                  = 0
read(15, "Xcur\20\0\0\0\0\0\1\0\1\0\0\0\2\0\375\377 \0\0\0\34\0\0\0$\0\0\0"..., 4096) = 4096
read(15, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 4096) = 64
close(15)                               = 0
munmap(0x7f64c5dea000, 4096)            = 0
poll([{fd=5, events=POLLIN|POLLOUT}], 1, -1) = 1 ([{fd=5, revents=POLLOUT}])
writev(5, [{"<\5\2\0\351\1\300\0055 \4\0\352\1\300\5\247\0\0\0 \0 \0007\0\4\0\353\1\300\5"..., 4552}, {NULL, 0}, {"", 0}], 3) = 4552
poll([{fd=5, events=POLLIN}], 1, -1)    = 1 ([{fd=5, revents=POLLIN}])
read(5, "\16\0\212\5\342\1\300\5\0\0>\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 4096) = 64
read(5, 0x33ec424, 4096)                = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=5, events=POLLIN|POLLOUT}], 1, -1) = 1 ([{fd=5, revents=POLLOUT}])
writev(5, [{"<\5\2\0\361\1\300\0057 \5\0\362\1\300\5\360\1\300\5\0\200\0\0\1\0\0\0>\1\7\0"..., 60}, {NULL, 0}, {"", 0}], 3) = 60
poll([{fd=5, events=POLLIN}], 1, -1)    = 1 ([{fd=5, revents=POLLIN}])
read(5, "\16\0\216\5$\0\300\5\0\0>\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 4096) = 64
read(5, 0x33ec424, 4096)                = -1 EAGAIN (Resource temporarily unavailable)
open("/usr/share/fonts/truetype/ubuntu-font-family/Ubuntu-B.ttf", O_RDONLY) = 15
fcntl(15, F_SETFD, FD_CLOEXEC)          = 0
fstat(15, {st_mode=S_IFREG|0644, st_size=312932, ...}) = 0
mmap(NULL, 312932, PROT_READ, MAP_PRIVATE, 15, 0) = 0x7f64ae28c000
close(15)                               = 0
futex(0x2d01d6c, FUTEX_WAKE_PRIVATE, 2147483647) = 0
open("/etc/localtime", O_RDONLY)        = 15
fstat(15, {st_mode=S_IFREG|0644, st_size=2309, ...}) = 0
fstat(15, {st_mode=S_IFREG|0644, st_size=2309, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f64c5dea000
read(15, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\10\0\0\0\10\0\0\0\0"..., 4096) = 2309
lseek(15, -1467, SEEK_CUR)              = 842
read(15, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\t\0\0\0\t\0\0\0\0"..., 4096) = 1467
close(15)                               = 0
munmap(0x7f64c5dea000, 4096)            = 0
mmap(NULL, 2147483648, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = -1 ENOMEM (Cannot allocate memory)
--- SIGSEGV (Segmentation fault) @ 0 (0) ---
+++ killed by SIGSEGV +++
Speicherzugriffsfehler

</code>

Does anybody know, how this was caused?

**Comment by tea** [comment]
Can you provide a stack backtrace for a e.debug run?

**Comment by dennis605** [comment]
HI tea, thanx for reply,

i solved it my way. I think there was a prob with patching the fles "get_externals_linux.sh" has downloaded. 2 packages haven't been loaded correct.
Now almost everything works fine, but when i try to open webpreview 2 times, i got this error:
ASSERTION FAILED: vptr() == JSGlobalData::jsArrayVPtr
(../JavaScriptCore/runtime/JSArray.cpp:242 virtual JSC::JSArray::~JSArray()) 
Speicherzugriffsfehler

Can you help me solving this?

Thanx

dennis605

**Comment by tea** [comment]
I see you opened another issue for webkit problem, let's keep discussion about it in that issue (I'll see if I can reproduce/fix it)

**Comment by dennis605** [comment]
You're right, i just wanted to answer. We'll continue discussing in the other post.

**Comment by dennis605** [comment]
Sorry for posting in this issue again, but my other issue is closed somehow. Maybe i clicked "Comment and close" by mistake, but it should not be closed. Can i reopen it?
